plugin/evm/vm.go

```diff
@@ -933,13 +949,64 @@ func (vm *VM) SetState(_ context.Context, state snow.State) error {
 }
 
 // initBlockBuilding starts goroutines to manage block building
-func (vm *VM) initBlockBuilding() {
+func (vm *VM) initBlockBuilding() error {
 	// NOTE: gossip network must be initialized first otherwise ETH tx gossip will not work.
 	gossipStats := NewGossipStats()
 	vm.gossiper = vm.createGossiper(gossipStats)
 	vm.builder = vm.NewBlockBuilder(vm.toEngine)
 	vm.builder.awaitSubmittedTxs()
 	vm.Network.SetGossipHandler(NewGossipHandler(vm, gossipStats))
+
+	ethTxPool, err := NewGossipEthTxPool(vm.txPool)
+	if err != nil {
+		return err
+	}
+	vm.shutdownWg.Add(1)
+	go ethTxPool.Subscribe(vm.shutdownChan, &vm.shutdownWg)
+
+	atomicMempool, err := NewGossipAtomicMempool(vm.mempool)
+	if err != nil {
+		return err
+	}
+	vm.atomicMempool = atomicMempool
+
+	ethTxGossipHandler := gossip.NewHandler[GossipEthTx, *GossipEthTx](ethTxPool, vm.codec, message.Version)
+	ethTxGossipClient, err := vm.router.RegisterAppProtocol(0x0, ethTxGossipHandler)
+	if err != nil {
+		return err
+	}
+	vm.ethTxGossipClient = ethTxGossipClient
+
+	atomicTxGossipHandler := gossip.NewHandler[GossipAtomicTx, *GossipAtomicTx](atomicMempool, vm.codec, message.Version)
+	atomicTxGossipClient, err := vm.router.RegisterAppProtocol(0x1, atomicTxGossipHandler)
+	if err != nil {
+		return err
+	}
+	vm.atomicTxGossipClient = atomicTxGossipClient
+
+	vm.ethTxGossiper = gossip.NewGossiper[GossipEthTx, *GossipEthTx](
+		ethTxPool,
+		vm.ethTxGossipClient,
+		vm.networkCodec,
+		message.Version,
+		pullTxsGossipSize,
+		pullTxsFrequency,
+	)
+	vm.shutdownWg.Add(1)
+	go vm.ethTxGossiper.Pull(vm.shutdownChan, &vm.shutdownWg)
+
+	vm.atomicTxGossiper = gossip.NewGossiper[GossipAtomicTx, *GossipAtomicTx](
+		atomicMempool,
+		vm.atomicTxGossipClient,
+		vm.networkCodec,
+		message.Version,
+		pullTxsGossipSize,
+		pullTxsFrequency,
+	)
+	vm.shutdownWg.Add(1)
+	go vm.atomicTxGossiper.Pull(vm.shutdownChan, &vm.shutdownWg)
```

Comment:
same comment on whether we should make these constructors one line constructors and pass in the necessary extra params